### PR TITLE
[EWS O365] Fixed issue with invalid characters in the From header

### PR DESCRIPTION
**Pull request by mmhw**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-42219)

## Description
Fixed an issue where attachments with invalid characters in the From header were not being processed correctly.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py [hunk 1]
```diff
@@ -2128,6 +2130,45 @@ def handle_attached_email_with_incorrect_message_id(attached_email: Message):
...
+        attached_email (EmailMessage): attached email object.
+
+    Returns:
+        Message: attached email object.
+    """
+    for i in range(len(attached_email._headers)):
+        if attached_email._headers[i][0].lower() == "from":
+            from_header = attached_email._headers[i][0]
+            value = attached_email._headers[i][1]
+            demisto.debug(f'Handling From header, {value=}.')
```
Comment (michal-dagan) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py:2149]:
```suggestion
    for i, (header_name, header_value) in enumerate(attached_email._headers):
        if header_name.lower() == "from":
            demisto.debug(f'Handling From header, value={header_value}.')
```

File: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py [hunk 2]
```diff
@@ -2128,6 +2130,45 @@ def handle_attached_email_with_incorrect_message_id(attached_email: Message):
...
+            try:
+                new_value = parser.get_address_list(value)[0].value
+                new_value = new_value.replace("\n", " ").replace("\r", " ").strip()
+                if value != new_value:
+                    # If the From header was fixed in the context of this function
+                    # the header will be replaced in _headers list
+                    attached_email._headers.pop(i)
+                    attached_email._headers.insert(i, (from_header, new_value))
+                    demisto.debug(f'From header fixed, new value: {new_value}')
```
Comment (michal-dagan) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py:2158]:
```suggestion
    if header_value != new_value:
                    # Update the 'From' header with the corrected value
                    attached_email._headers[i] = (header_name, new_value)
                    demisto.debug(f'From header fixed, new value: {new_value}')
                    
```

File: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py [hunk 3]
```diff
@@ -2128,6 +2130,45 @@ def handle_attached_email_with_incorrect_message_id(attached_email: Message):
...
+
+            except Exception as e:
+                # The function is designed to handle a specific format error for the From header
+                # as explained in the docstring.
+                # That being said, we do expect the header to be in a known format.
+                # If this function encounters a header format which is not in the known format and can't be fixed,
+                # the header will be ignored completely to prevent crashing the fetch command.
+                demisto.debug(f"Invalid Error: {e}")
```
Comment (michal-dagan) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py:2166]:
```suggestion
                demisto.debug(f"Error processing From header: {e}")
```

**Review by michal-dagan — APPROVED** [review]
Great!

File: Packs/MicrosoftExchangeOnline/ReleaseNotes/1_5_9.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### EWS O365
+
+Fixed an issue where attachments with invalid characters in the From header were not being processed correctly.
```
Comment (DeanArbel) [inline Packs/MicrosoftExchangeOnline/ReleaseNotes/1_5_9.md:6]: Please be more specific - the invalid characters will now be replaced by white space character as to allow processing these mails.

**Review by DeanArbel — APPROVED** [review]
Looks good, see my comment.

**Pull request merged**